grpc/src/main/java/com/linecorp/armeria/internal/grpc/ArmeriaMessageDeframer.java

```diff
@@ -177,7 +179,8 @@ public int hashCode() {
     private boolean endOfStream;
 
     @Nullable
-    private CompositeByteBuf unprocessed;
+    private Queue<ByteBuf> unprocessed;
```

Comment:
This is really surprising - I feel like you've reimplemented `CompositeByteBuf`. Is this not a use case for that class?

Also, can you compare against the original code with a Netty snapshot? By total coincidence looks like there have been major optimizations lately

https://github.com/netty/netty/commits/4.1/buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

Reply:
`CompositeByteBuf` is a general purpose buffer implementation that has to maintain a lot more information than the implementation in this PR. It's much simpler because we always read from the beginning and we do not have to worry about managing the component list after `readerIndex` advances beyond a component.